packages/database/src/models/User.ts

```diff
@@ -273,27 +273,32 @@ export class User extends Model {
 
     const restrictUsersToFacilities = await Setting.get('auth.restrictUsersToFacilities');
     const hasLoginPermission = await this.hasPermission('login', 'Facility');
+    const hasAllNonSensitiveFacilityAccess = !restrictUsersToFacilities || hasLoginPermission;
+
+    const sensitiveFacilities = await Facility.count({ where: { isSensitive: true } });
+    if (hasAllNonSensitiveFacilityAccess && sensitiveFacilities === 0)
+      return CAN_ACCESS_ALL_FACILITIES;
 
     // Get user's linked facilities
     if (!this.facilities) {
       await this.reload({ include: 'facilities' });
     }
-    const userLinkedFacilities = this.facilities?.map(({ id, name }) => ({ id, name })) ?? [];
+    const userFacilities = this.facilities?.map(({ id, name }) => ({ id, name })) ?? [];
```

Comment:
Eeek sorry again with the naming nits, but I want to avoid this being read as containing user_facilities records too 
```suggestion
    const explicitlyAllowedFacilities = this.facilities?.map(({ id, name }) => ({ id, name })) ?? [];
```
Honestly don't love the name, but have tried a few and that's my best shot